amethyst_gltf/src/format/mesh.rs

```diff
@@ -202,20 +203,52 @@ fn calculate_tangents(
     tex_coords: &[TexCoord],
     indices: &Indices,
 ) -> Vec<Tangent> {
-    let mut tangents = vec![Tangent([0.0, 0.0, 0.0, 0.0]); positions.len()];
-    let num_faces = indices.len().unwrap_or_else(|| positions.len()) / 3;
-    mikktspace::generate_tangents(
-        &|| 3,
-        &|| num_faces,
-        &|face, vert| &positions[indices.map(face, vert)].0,
-        &|face, vert| &normals[indices.map(face, vert)].0,
-        &|face, vert| &tex_coords[indices.map(face, vert)].0,
-        &mut |face, vert, tangent| {
+    struct TangentsGeometry<'a> {
```

Comment:
Heya @MilanVasko, thank you for this! Could you move `TangentsGeometry` and the trait impl outside the function? Shall be good to take the PR then :v:

Reply:
Hello, sure thing. Commit is coming in a couple of minutes. 😄